### ISSUE TITLE
Multiple Instances of Gitter start automatically. Can't uninstall.

### ISSUE THREAD

**Issue by R-Lefebvre** [gitterHQ/gitter#1377]
Hi,

Running Windows 10.

I've sort of been using Gitter for about a year with a software project team.  I mostly have not used it because it's been so unreliable on Windows.  Used to be that it would crash with nearly every notification.  I'd have to restart it manually many times a day.  That's when I started just not using it, and not using that comms channel to communicate with the team.

At some point, there was a big update, and the crash on notification problem stopped.  But I think it was from that point on, I now had 2 instances of Gitter opening every time I boot.  No idea why.  I continued to just ignore it, turned off notifications,  and just minimized both instances on every boot.

Yesterday, our team tried to do a Gitter/Skype integration/bridge.  I don't know if that is related, but suddenly I had 3 instances running.  And it turned notifications on.  Now with every incoming message, I got 1 notification from Skype, and 3 from Gitter.

Had enough, I uninstalled the program.

After rebooting, I still have 2 instances of Gitter starting.  Gitter no longer appears in my Uninstall Program Menu.

Would be happy to use Gitter if it worked reliably on Windows.  I was going to uninstall/reinstall to fix the multiple instances problem. But that didn't work.  Any ideas?

**Comment by SirAssembly** [comment]
Robert,

Sorry to hear about your issues. Have you tried stopping the program in Task Manager's startup tab?

http://anewdomain.net/2015/07/26/how-to-disable-startup-programs-in-windows-10-and-speed-up-load-time/

To my understanding multiple instances of the same program may reside here (thereby cause redundant boots) and can be disabled respectively (possibly separate publishers? Maybe one instance from separate install that went somewhere funky?). You can also "Open File location" to see where it is booting from if you have it uninstalled from your official Program Files directory but it still exists.

I have never officially posted so please forgive any missed formalities. Let me know if this fixes the issue! What did you use to bridge? Sameroom?

Regards,
Josh Ward

**Comment by MadLittleMods** [comment]
@R-Lefebvre Not sure how I missed this issue but @SirAssembly comment appeared in my inbox.

To clarify, every time you launch Gitter, it will be a separate instance, this is because our update process requires the old and new version to run. 

It looks like you are suffering from https://github.com/gitterHQ/desktop/issues/114 but that was [fixed in v3.0.2](https://github.com/gitterHQ/desktop/blob/b2c0288f90cc9813648505f9e999ecbaa02657d2/CHANGELOG.md#302---2016-2-26). Which version of the desktop client are you using? That code is still in place today though: https://github.com/gitterHQ/desktop/blob/b2c0288f90cc9813648505f9e999ecbaa02657d2/nwapp/index.js#L61 but perhaps some permissions are being denied when trying to get rid of it. You can [setup logging](https://github.com/gitterHQ/desktop#enabling-logging-on-windows) and see if the "Failed to cleanup legacy startup file" message is in there or just see if that file still exists and remove it.

The third instance of Gitter starting seems a bit weird but perhaps you are launching Gitter when you start the bridge or something and the multiple instance thing is happening to you.

Startup + Startup(duplicate bug) + extra launch from bridge

If you want to delete the rest of Gitter:

> On Windows, the app has files in:
> 
>  - `C:\Users\YOUR_USER\AppData\Local\Programs\Gitter`: App install location
>  - `C:\Users\YOUR_USER\AppData\Local\Gitter`: Config, see https://github.com/nwjs/nw.js/wiki/app#datapath
> 
> Unfortunately, pretty sure we leave behind files especially if you have done an update.
> 
> https://github.com/gitterHQ/gitter/issues/1366#issuecomment-240568724

**Comment by MadLittleMods** [comment]
@R-Lefebvre Do you need any further info/help?

**Comment by R-Lefebvre** [comment]
Hi, sorry for not responding.

It's hard to say if I need more help still.  :)  The status right now, is that I am magically back to 1 instance running.  I didn't do anything, a few days after I posted this I think, I noticed there is only one running.  I didn't do anything to change the situation.  I am not sure if you pushed some updated which fixed it or what.

Gitter is still not registered in my Windows program list so I can't uninstall it.  But all of the program files are still in my Programs/Gitter directory. Actually, just in case that is a clue for you,  I'll point out that my file structure is not standard.  My C: drive where Windows is installed is a small SSD.  I try to install most programs on my E: drive which is a standard hard drive.  I still have C:/Program Files and then Gitter is actually installed in E:/Programs/Gitter.  Maybe your software is expecting everything to be in the standard location, and that triggered something going wrong?

I still need to use Gitter to interface with my team, it's basically working now, so I'm happy to leave well enough alone for now.
